pkg/services/ngalert/state/manager.go

```diff
@@ -164,18 +163,53 @@ func (st *Manager) Get(orgID int64, alertRuleUID, stateId string) *State {
 	return st.cache.get(orgID, alertRuleUID, stateId)
 }
 
-// ResetStateByRuleUID deletes all entries in the state manager that match the given rule UID.
-func (st *Manager) ResetStateByRuleUID(ctx context.Context, ruleKey ngModels.AlertRuleKey) []*State {
+// ResetStateByDeletedRuleUID deletes all entries in the state manager that match the given rule UID.
+func (st *Manager) ResetStateByDeletedRuleUID(ctx context.Context, ruleKey ngModels.AlertRuleKey) []*State {
+	return st.resetStateByRuleUID(ctx, ruleKey, nil, ngModels.StateReasonDeleted)
+}
+
+// ResetStateByPausedRuleUID deletes all entries in the state manager that match the given rule UID.
+func (st *Manager) ResetStateByPausedRuleUID(ctx context.Context, ruleKey ngModels.AlertRuleKey, rule *ngModels.AlertRule) []*State {
+	return st.resetStateByRuleUID(ctx, ruleKey, rule, ngModels.StateReasonPaused)
+}
```

Comment:
Not sure I understand why there are two methods that do the same operation. I think this should be reduced to ResetStateByRuleUID that accepts reason as parameter.

Reply:
There are 2 methods to abstract `schedule.go` from slight method differences. If we prefer only one function I can change it.